### PR TITLE
Add a method to generate the error in f_diffuse given an error in Mstar

### PR DESCRIPTION
**Pull request by profxj**
This attempts to capture uncertainty in `f_d` from uncertainties
in rho_*, dominated by IMF.

Note that the default `f_d` assumes Salpeter and in some ways
is an "extreme" value with Chabrier the other bound

### CODE REVIEW

**Review by SunilSimha — CHANGES_REQUESTED** [review]
Requesting one change.

File: frb/dm/igm.py [hunk 1]
```diff
@@ -147,71 +147,98 @@ def z_from_DM(DM, cosmo=defs.frb_cosmo, coord=None, corr_nuisance=True):
...
+    """
+    # Get comoving baryon mass density
+    rho_b = cosmo.Ob0 * cosmo.critical_density0.to('Msun/Mpc**3')
+
+    # Dense components
+    rho_Mstar = avg_rhoMstar(z, remnants=True)
+    if perturb_Mstar is not None:
+        rho_Mstar *= perturb_Mstar
+        
+    rho_ISM = avg_rhoISM(z, cosmo=cosmo)
```
[the remark this answers is not in the document]
Comment (profxj) [reply frb/dm/igm.py:192]: good catch!  fixed

**Comment by profxj** [comment]
tests running successfully on my laptop:

- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========== 61 passed, 3 skipped, 482 warnings in 61.67s (0:01:01) ============

**Review by SunilSimha — APPROVED** [review]
I made a small bug fix. Looks good to merge now.

**Comment by profxj** [comment]
merging despite the test failures that
are the result of `linetools` which I will
endeavor to deprecate in a future PR

**Pull request merged**